src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -135,43 +247,167 @@ public void run() {
             return null;
         }
+
+        try {
+            if(!clusterColorControl.discoverAttributes(false).get()) {
+                logger.warn("{}: Cannot determine whether device endpoint supports RGB color. Assuming it does by now (checking again later)", endpoint.getEndpointAddress());
+            }
+            else if(clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTHUE)) {
+                logger.debug("{}: Device endpoint supports HUE/SATURATION color", endpoint.getEndpointAddress());
+            }
+            else if(clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTX)) {
+                logger.debug("{}: Device endpoint supports XY color", endpoint.getEndpointAddress());
+            }
+            else {
+                logger.debug("{}: Device endpoint does not support RGB color (probably only color temperature)", endpoint.getEndpointAddress());
+                return null;
+            }
+        }
+        catch(InterruptedException | ExecutionException e) {
+                logger.warn("{}: Exception checking whether device endpoint supports RGB color. Assuming it does by now (checking again later)", endpoint.getEndpointAddress(), e);
+        }
+
         return createChannel(thingUID, endpoint, ZigBeeBindingConstants.CHANNEL_COLOR_COLOR,
                 ZigBeeBindingConstants.ITEM_TYPE_COLOR, "Color");
     }
 
+    protected void updateOnOff(OnOffType onoff) {
+        HSBType oldHSB = currentHSB;
+        updateChannelState(onoff);
+        logger.debug("{}: OnOff updated. {} -> {}", endpoint.getEndpointAddress(), oldHSB.toString(), onoff.toString());
+    }
+
+    protected void updateBrightness(PercentType brightness) {
+        HSBType oldHSB = currentHSB;
+        HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
+        currentHSB = newHSB;
+        updateChannelState(brightness);
+        updateChannelState(newHSB);
+        logger.debug("{}: Brightness updated. {} -> {}", endpoint.getEndpointAddress(), oldHSB.toString(), newHSB.toString());
+    }
+
+    protected void updateColorHSB(DecimalType hue, PercentType saturation) {
+        HSBType oldHSB = currentHSB;
+        HSBType newHSB = new HSBType(hue, saturation, oldHSB.getBrightness());
+        currentHSB = newHSB;
+        updateChannelState(newHSB);
+        logger.debug("{}: Color updated. {} -> {}", endpoint.getEndpointAddress(), oldHSB.toString(), newHSB.toString());
+    }
+
+    protected void updateColorXY(PercentType x, PercentType y) {
+        logger.debug("{}: Update Color XY. {}, {}", endpoint.getEndpointAddress(), x.toString(), y.toString());
+        // float hsb[] = ColorHelper.xy2hsv(lastX, lastY);
+        float rgb[] = ColorHelper.xy2rgb(x.floatValue() / 100.0f, y.floatValue() / 100.0f);
+        float hsb[] = ColorHelper.rgb2hsv(rgb[0], rgb[1], rgb[2]);
```

Comment:
There's already a method to do this in the HSBType in ESH. I wonder if we shouldn't look to add new methods to this class rather than add a separate ColorHelper class to ZigBee (ie add the XY methods to the existing class)?

Reply:
Should then be in the HSBType. Will move it there.